### ISSUE TITLE
<Appear> tags don't show up in print or presenter mode

### ISSUE THREAD

**Issue by bebraw** [FormidableLabs/spectacle#294]
Example: http://presentations.survivejs.com/webpack-from-apprentice-to-journeyman/#/?export&print -> cmd-p -> no content for `Appear` tags.

**Comment by bebraw** [comment]
http://presentations.survivejs.com/webpack-from-apprentice-to-journeyman/#/?overview is the same.

I checked `Appear` and it actually has a check for this. Perhaps routing has changed breaking that.

#293 might be related to this.

**Comment by kstratis** [comment]
#293 is definitely the same thing. However you probably mean presenter mode instead of print mode.

Even stranger though is the fact that your presentation example works exactly as it should while the example in the src tree doesn't...

**Comment by bebraw** [comment]
There's definitely something weird going on. First it didn't work and now it does. Some race condition thingy?

I don't mind closing this in favor of #293.

**Comment by kstratis** [comment]
I don't mind either. I've mentioned both issues on gitter but no updates yet. 

I'll try to figure out what's going on here but I believe I'll have to dig deep for this one and it may take some time.

Any help would be much appreciated.

**Comment by kstratis** [comment]
@bebraw What we could do in the meantime is to check the part of your source code against the one on the main example. There may be something in there...

**Comment by bebraw** [comment]
I think I got it (initial reason anyway). It's the **first loading** that doesn't work. After that it seems to be fine.

I have a theory too. The current check is done **only** in `componentWillReceiveProps`. We probably want to handle it in `componentWillMount` too capture the first phase.

**Comment by kstratis** [comment]
Which file are you talking about?

My bet is at the `manager.js` file in the `_checkFragments` function.
But I'm not sure yet...

Did you come up with a solution?

**Comment by kstratis** [comment]
@bebraw I haven't found the culprit yet but I've made some progress.

Here's what I know so far:

**The issue appears in the master branch only**. 

If you checkout the spectacle-boilerplate which uses the npm version of spectacle you get no such error. That's why the slides you shared in your post work correctly.

However if you run the spectacle from the master branch the issue is there. Using a "binary search" technique I managed to spot the PRs responsible for this behaviour even though I couldn't pin down where exactly the error is cause there is a vast amount of files and lines changed.

* #280 introduced  part of the issue: While in the presenter mode, the `<Appear>` tags will properly show up but if you progress to the next slide and then go back they suddenly disappear.
* #281 completed the last part of the issue: While in the presenter mode, the `<Appear>` tags will now disappear from the current slide whether we traverse forth or backwards.

I believe that now it should be easier for someone to pick it up...
